lib/facter/warranty.rb

```diff
@@ -120,7 +120,7 @@ def create_lenovo_warranty_cache(cache)
     end
 
     # refresh cache daily
-    if File.exists?(cache_file) and Time.now < File.stat(cache_file).mtime + 86400 * 1
```

Comment:
This seems to have been fixed in puppet 8.3.0.

https://www.puppet.com/docs/puppet/8/release_notes_puppet.html#enhancements_puppet_x-8-3-0-PUP-11945